substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -1716,6 +1862,395 @@ static Object getMulticastInterface(int fd, int opt) throws SocketException {
+        //  1138      int index;
+        CIntPointer index_Pointer = StackValue.get(CIntPointer.class);
+        //  1139
+        //  1140      if (ni_indexID == NULL) {
+        //  1141          jclass c = (*env)->FindClass(env, "java/net/NetworkInterface");
+        //  1142          CHECK_NULL(c);
+        //  1143          ni_indexID = (*env)->GetFieldID(env, c, "index", "I");
+        //  1144          CHECK_NULL(ni_indexID);
+        //  1145      }
+        //  1146      index = (*env)->GetIntField(env, value, ni_indexID);
+        index_Pointer.write(Util_java_net_NetworkInterface.fromNetworkInterface(value).index);
+        //  1147
+        //  1148      if (JVM_SetSockOpt(fd, IPPROTO_IPV6, IPV6_MULTICAST_IF,
+        //  1149                         (const char*)&index, sizeof(index)) < 0) {
+        if ( VmPrimsJVM.JVM_SetSockOpt(fd, NetinetIn.IPPROTO_IP(), NetinetIn.IPV6_MULTICAST_IF(), (CCharPointer) index_Pointer, SizeOf.get(CIntPointer.class)) <0) {
+            //  1150          if (errno == EINVAL && index > 0) {
+            if (Errno.errno() == Errno.EINVAL() && index_Pointer.read() > 0) {
+                //  1151              JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException",
+                //  1152                  "IPV6_MULTICAST_IF failed (interface has IPv4 "
+                //  1153                  "address only?)");
+                throw new SocketException("IPV6_MULTICAST_IF failed (interface has IPv4 address only?");
+            } else {
+                //  1155              NET_ThrowByNameWithLastError(env, JNU_JAVANETPKG "SocketException",
+                //  1156                             "Error setting socket option");
+                throw new SocketException("Error setting socket option");
+            }
+            //  1158          return;
+            /* Unreachable. */
+        }
+    }
+    //  1162  #endif /* AF_INET6 */
+    // @formatter:on
+
+    /* Do not re-format commented out code: @formatter:off */
+    //  1181  /*
+    //  1182   * Set outgoing multicast interface designated by an InetAddress.
+    //  1183   * Throw exception if failed.
+    //  1184   */
+    //  1185  #ifdef AF_INET6
+    //  1186  static void mcast_set_if_by_addr_v6(JNIEnv *env, jobject this, int fd, jobject value) {
+    static void mcast_set_if_by_addr_v6(int fd, InetAddress valueArg) throws SocketException {
+        //  1187      static jclass ni_class;
+        //  1188      if (ni_class == NULL) {
+        //  1189          jclass c = (*env)->FindClass(env, "java/net/NetworkInterface");
+        //  1190          CHECK_NULL(c);
+        //  1191          ni_class = (*env)->NewGlobalRef(env, c);
+        //  1192          CHECK_NULL(ni_class);
+        //  1193      }
+        //  1194
+        //  1195      value = Java_java_net_NetworkInterface_getByInetAddress0(env, ni_class, value);
+        NetworkInterface value;
+
+        value = Target_java_net_NetworkInterface.getByInetAddress0(valueArg);
```

Comment:
Why does 1195 translate to 3 lines?  Because you can not *believe* that they take an `InetAddress` formal parameter and reassign (!) it with a `NetworkInterface` value?  One line, (maybe with a comment) would be enough.